### PR TITLE
Add redis client

### PR DESCRIPTION
**Pull request by adelinag08**
Part of #35

### CODE REVIEW

File: src/Storage/Base/IRedisSubscriberClient.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+using System;
+using StackExchange.Redis;
+
+namespace Snd.Sdk.Storage.Base;
+
+/// <summary>
+/// Interface for Redis subscriber operations.
+/// </summary>
+public interface IRedisSubscriberClient
```
Comment (s-vitaliy) [inline src/Storage/Base/IRedisSubscriberClient.cs:9]: Can we use `IPublisher` and `ISubscriber` from the `Reactive.Streams` library? https://github.com/reactive-streams/reactive-streams-dotnet 

In this case [we can easily integrate that with Akka.NET](https://getakka.net/articles/streams/integration.html#integrating-with-reactive-streams) if we need.

File: src/Storage/Cache/RedisClient.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    {
+        var db = redis.GetDatabase();
+        return db.StringGet(key);
+    }
+
+    /// <inheritdoc />
+    public List<string> MultiGet(List<string> keys)
+    {
+        var db = redis.GetDatabase();
+        var redisKeys = keys.ConvertAll(k => (RedisKey)k).ToArray();
```
Comment (s-vitaliy) [inline src/Storage/Cache/RedisClient.cs:73]: Are, you sure, we need only type cast here?

**Review by s-vitaliy — COMMENTED** [review]
Added some comments to implementation.
Also, does this library has asynchronous client?

File: src/Storage/Cache/RedisClient.cs (hunk 2 above)
Comment (adelinag08) [reply src/Storage/Cache/RedisClient.cs]: yes, I tested and it works as expected

File: src/Storage/Base/IRedisSubscriberClient.cs (hunk 1 above)
Comment (adelinag08) [reply src/Storage/Base/IRedisSubscriberClient.cs]: I will do the Publisher/Subscriber implementation in another PR

File: src/Storage/Providers/RedisServiceProvider.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+        var options = new ConfigurationOptions()
+        {
+            EndPoints = { { redisConfiguration.Host, redisConfiguration.Port } },
+            Password = redisConfiguration.Password,
+            Ssl = redisConfiguration.UseSsl,
+            DefaultDatabase = redisConfiguration.DatabaseNumber
+        };
+
+        services.AddSingleton(typeof(IConnectionMultiplexer), ConnectionMultiplexer.Connect(options));
```
Comment (s-vitaliy) [inline src/Storage/Providers/RedisServiceProvider.cs:34]: Discussed by voice: if created like that, the `IConnectionMultiplexer` instance will not be disposed by DI container, it's better to use factory method. Also, see: https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection

**Review by s-vitaliy — APPROVED** [review]
LGTM with minor comment

**Pull request merged**